src/mame/drivers/rastersp.cpp

```diff
@@ -436,7 +614,128 @@ WRITE_LINE_MEMBER( rastersp_state::vblank_irq )
 		update_irq(IRQ_VBLANK, ASSERT_LINE);
 }
 
+WRITE_LINE_MEMBER( rastersp_state::duart_irq )
+{
+	update_irq(IRQ_UART, state);
+}
+
 
+WRITE_LINE_MEMBER( fbcrazy_state::trackball_rts )
+{
+	m_trackball_enabled = state;
+}
+
+/*
+	Mimic trackball unit. This uses a small PIC based interface board
+	which converts the trackball movement into RS232 data.
+	The data format is as follows:
+	Byte 0 : 0100abAB bits 7 & 6 = 01 signifies 1st byte
+	Byte 1 : 00CDEFGH bits 7 & 6 = 00 signifies 2 or 3rd byte (not identified)
+	Byte 2 : 00cdefgh
+
+	The trackball X movement is ABCDEFGH and Y movement abcdefgh
+	We do not have a PCB nor a dump of the PIC :(
+*/
+TIMER_CALLBACK_MEMBER(fbcrazy_state::trackball_timer)
+{
+	if (m_trackball_enabled)
+	{
+		if (m_trackball_ctr == 0)
+		{
+			uint16_t newxpos, newypos;
+
+			newxpos = ioport("TRACK_X")->read();
+			newypos = ioport("TRACK_Y")->read();
+
+			int diffx, diffy;
+			diffx = newxpos - m_oldxpos;
+			diffy = newypos - m_oldypos;
+
+			if (diffx<-3900)
+			{
+				diffx += 4096;
+			}
+			else if (diffx>3900)
+			{
+				diffx -= 4096;
+			}
+			if (diffx>127)
+			{
+				diffx = 127;
+			}
+			else if (diffx <-127)
+			{
+				diffx = -127;
+			}
+
+			if (diffy<-3900)
+			{
+				diffy += 4096;
+			}
+			else if (diffy>3900)
+			{
+				diffy -= 4096;
+			}
+			if (diffy>127)
+			{
+				diffy = 127;
+			}
+			else if (diffy <-127)
+			{
+				diffy = -127;
+			}
+
+			m_oldxpos = newxpos;
+			m_oldypos = newypos;
+
+			//Format movement into 3 bytes as described above for tx
+			m_trackball_data[0] = 0x40 | ((diffy>>4)&0x0c) | ((diffx>>6)&0x03);
+			m_trackball_data[1] = diffx & 0x3f;
+			m_trackball_data[2] = diffy & 0x3f;
+		}
+
+		if (m_trackball_ctr < 36)
+		{
+			//Transmitting data
+			uint8_t bitnum = m_trackball_ctr % 12 ;
+			if (bitnum == 0)
+			{
+				//start bit
+				m_duart->rxb_w(0);
+			}
+			else if (bitnum < 9)
+			{
+				//Data bits
+				if (m_trackball_data[(m_trackball_ctr/12)]&(1<<(bitnum-1)))
+				{
+					m_duart->rxb_w(1);
+				}
+				else
+				{
+					m_duart->rxb_w(0);
+				}
```

Comment:
There are a lot of parentheses and it looks like your space bar was broken for a bit.  Couldn’t you just do `m_duart->rxb_w(BIT(m_trackball_data[m_trackball_ctr / 12], bitnum - 1));` and save some lines?